### ISSUE TITLE
0.9.1: Undefined variable "instance_datadir_config" at /etc/puppet/environments/staging3/modules/elasticsearch/manifests/instance.pp:239

### ISSUE THREAD

**Issue by mfournier** [elasticsearch/puppet-elasticsearch#284]
Removing the "path" section from the config hash passed to `elasticsearch::instance` makes the error go away. Here's what it contained:
```
      'path'      => {
        'home' => '/usr/share/elasticsearch',
      },
```

**Comment by electrical** [comment]
Thank you for reporting this. I found the issue and will push a fix for this tomorrow.
The issue is that i've no 'else' part at https://github.com/elasticsearch/puppet-elasticsearch/blob/master/manifests/instance.pp#L183-L185

**Comment by mfournier** [comment]
Yes, I figured that out :) Thanks for fixing this so quickly ? Any ETA for a 0.9.2 ?

For the record, I'm working around this by starting logstash with `'LS_JAVA_OPTS' => '"-Des.path.home=/usr/share/elasticsearch"'` (I'm sharing the elasticsearch config with logstash so that it's embedded client can automatically benefit from the same settings).

**Comment by electrical** [comment]
I'm going to try to do a release tomorrow, otherwise I'll be doing that while I'm in SF at Elasticon.

**Comment by electrical** [comment]
I'm currently working on a fix including a test to catch it for the future.

**Comment by electrical** [comment]
@mfournier https://github.com/elasticsearch/puppet-elasticsearch/pull/289 should fix this issue.
Will be merging it today and trying to do a release today.

**Comment by electrical** [comment]
I've released 0.9.2 just a few minutes ago. Hope this completely solves your issue.
